src/ChluIPFS.js

```diff
@@ -167,37 +167,98 @@ class ChluIPFS {
-        const buffer = dagNode.Data;
+        const buffer = dagNode.data;
         const messages = protons(require('../src/utils/protobuf'));
         const reviewRecord = messages.ReviewRecord.decode(buffer);
-        // TODO: validate
         return reviewRecord;
     }
 
-    async storeReviewRecord(reviewRecord){
+    async readReviewRecord(multihash, notifyUpdate = null) {
+        const reviewRecord = await this.getReviewRecord(multihash);
+        // TODO validate
+        if (notifyUpdate) this.findLastReviewRecordUpdate(multihash, notifyUpdate);
+        return reviewRecord;
+    }
+
+    async storeReviewRecord(reviewRecord, previousVersionMultihash = null){
         if (this.type !== constants.types.customer) {
             throw new Error('Not a customer');
         }
-        let buffer;
-        if (Buffer.isBuffer(reviewRecord)) {
-            buffer = reviewRecord;
-        } else if (typeof reviewRecord === 'object') {
-            const messages = protons(require('./utils/protobuf'));
-            buffer = messages.ReviewRecord.encode(reviewRecord);
-        } else {
-            throw new Error('Unrecognised reviewRecord type: either pass a protobuf encoded buffer or an object');
-        }
+        reviewRecord.orbitDb = this.getOrbitDBAddress();
+        const messages = protons(require('./utils/protobuf'));
+        const buffer = messages.ReviewRecord.encode(reviewRecord);
+        // TODO validate
         // write thing to ipfs
         const dagNode = await this.ipfs.object.put(buffer);
         const multihash = this.utils.multihashToString(dagNode.multihash);
         // Broadcast request for pin, then wait for response
         // TODO: handle a timeout and also rebroadcast periodically, otherwise new peers won't see the message
-        await new Promise(fullfill => {
+        let tasksToAwait = [];
+        tasksToAwait.push(new Promise(fullfill => {
             this.events.once(constants.eventTypes.pinned + '_' + multihash, () => fullfill());
             this.room.broadcast(this.utils.encodeMessage({ type: constants.eventTypes.wroteReviewRecord, multihash }));
+        }));
+        if (previousVersionMultihash) {
+            // This is a review update
+            tasksToAwait.push(this.setForwardPointerForReviewRecord(previousVersionMultihash, multihash));
+        }
+        await Promise.all(tasksToAwait);
+        return multihash;
+    }
+
+    async setForwardPointerForReviewRecord(previousVersionMultihash, multihash) {
+        this.logger.debug('Setting forward pointer for ' + previousVersionMultihash + ' to ' + multihash);
+        // TODO: verify that the update is valid
+        await new Promise(async fullfill => {
+            const address = this.getOrbitDBAddress();
+            this.events.once(constants.eventTypes.replicated + '_' + address, () => fullfill());
+            await this.db.set(previousVersionMultihash, multihash);
+            this.broadcastReviewUpdates();
```

Comment:
We should catch errors, and put in a TODO of retrying, if orbitDB doesn't do retrying.

Reply:
I'll check if orbit-db can throw when set is called, but I think it does not since what this does is adding a new operation to the operation log that is used by the orbit-db key value store. Then this new head of the log is replicated to other peers, but that is not part of the `set` call.

As for `broadcastReviewUpdates` that just asks other chlu nodes to replicate the orbit-db key value store. Periodically broadcasting it might be a good idea but I think it's just an optimization we can do later

Reply:
I checked the code for [orbit-db-kvstore](https://github.com/orbitdb/orbit-db-kvstore) and `set` cannot error out or fail so I don't think we need to catch errors